channels/1ds-post-js/src/HttpManager.ts

```diff
@@ -341,87 +317,79 @@ export class HttpManager {
 
             _self.serializeOfflineEvt = (evt) => {
-                return _serializer.getEventBlob(evt);
+                try {
+                    if (_serializer) {
+                        return _serializer.getEventBlob(evt);
+                    }
+                      
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+
+                }
+                return STR_EMPTY;
+                
             }
 
             _self.getOfflineRequestDetails = () => {
-                let payload = _serializer.createPayload(0, false, false, false, SendRequestReason.NormalSchedule, EventSendType.Batched);
-                return _buildRequestDetails(payload, _useHeaders);
-            }
+                try {
+                    let payload = _serializer && _serializer.createPayload(0, false, false, false, SendRequestReason.NormalSchedule, EventSendType.Batched);
+                    return _buildRequestDetails(payload, _useHeaders);
 
-            // Special internal method to allow the DebugPlugin to hook embedded objects
-            function _getSenderInterface(transports: TransportType[], syncSupport: boolean): IInternalXhrOverride {
-                let transportType: TransportType = TransportType.NotSet;
-                let sendPostFunc: SendPOSTFunction = null;
-                let lp = 0;
-                while (sendPostFunc == null && lp < transports.length) {
-                    transportType = transports[lp];
-                    if (transportType === TransportType.Xhr) {
-                        if (useXDomainRequest()) {
-                            sendPostFunc = _xdrSendPost;
-                        } else if (isXhrSupported()) {
-                            sendPostFunc = _xhrSendPost;
-                        }
-                    } else if (transportType === TransportType.Fetch && isFetchSupported(syncSupport) && (!syncSupport || (syncSupport && !_disableFetchKeepAlive))) {
-                        sendPostFunc = _fetchSendPost;
-                    } else if (_useBeacons && transportType === TransportType.Beacon && isBeaconsSupported()) {
-                        sendPostFunc = _beaconSendPost;
-                    }
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
 
-                    lp++;
                 }
+                return null;
+            }
 
-                if (sendPostFunc) {
-                    return {
-                        _transport: transportType,
-                        _isSync: syncSupport,
-                        sendPOST: sendPostFunc
-                    };
-                }
+            // Special internal method to allow the DebugPlugin to hook embedded objects
+            function _getSenderInterface(transports: TransportType[], syncSupport: boolean): _IInternalXhrOverride {
+                try {
+                    return _sendPostMgr && _sendPostMgr.getXhrInst(transports, syncSupport);
```

Comment:
Maybe we should rename the `getXhrInst` to `getSenderInst` as it doesn't really return "just" and XHR sender.